stan/math/rev/meta/ad_stack_type.hpp

```diff
@@ -0,0 +1,44 @@
+#ifndef STAN_MATH_REV_META_AD_STACK_TYPE_HPP
+#define STAN_MATH_REV_META_AD_STACK_TYPE_HPP
+
+#include <stan/math/prim/meta/is_eigen.hpp>
+#include <stan/math/prim/meta/plain_type.hpp>
+#include <stan/math/rev/core/ad_stack_allocator.hpp>
+
+namespace stan {
+namespace math {
+// forward declaration
+template <typename MatrixType>
+class AD_stack_matrix;
+}  // namespace math
+
+namespace internal {
+template <typename T, typename = void>
+struct AD_stack_type_impl {
+  using type = T;
```

Comment:
Should this be an error if it's not a specific type? Or does this handle all types?

Reply:
Yeah, I made it an arror to use this on an unsupported type.